packages/quantic/force-app/main/default/lwc/quanticResultAction/quanticResultAction.js

```diff
@@ -0,0 +1,149 @@
+import {LightningElement, api} from 'lwc';
+
+/** @typedef {import("coveo").Result} Result */
+
+/**
+ * The `QuanticResultAction` component allows the end user to perform a specific action on a result.
+ * @category Search
+ * @category Insight Panel
+ * @example
+ * <c-quantic-result-action result={result} event-name="attachtocase" icon-name="utility:chat" label="Attach result" selected selected-label="Detach result"></c-quantic-result-action>
+ */
+export default class QuanticResultAction extends LightningElement {
+  /**
+   * The label displayed in the tooltip of the result action button.
+   * @api
+   * @type {string}
+   */
+  @api label;
+  /**
+   * The label displayed in the tooltip when the result action button is selected.
+   * @api
+   * @type {string}
+   */
+  @api selectedLabel;
+  /**
+   * The name of the icon displayed inside the result action button.
+   * @api
+   * @type {string}
+   */
+  @api iconName;
+  /**
+   * Specifies whether the result action button is in selected state or not.
+   * @api
+   * @type {boolean}
+   */
+  @api selected;
+  /**
+   * Specifies whether the result action button is in loading state or not.
+   * @api
+   * @type {boolean}
+   */
+  @api loading;
+  /**
+   * The name of the event to be dispatched.
+   * @api
+   * @type {string}
+   */
+  @api eventName;
+  /**
+   * The result to perform the action on.
+   * @api
+   * @type {Result}
+   */
+  @api result;
+
+  /** @type {string} */
+  resultActionOrderClasses;
+  /** @type {boolean} */
+  _selected;
+  /** @type {boolean} */
+  _loading;
```

Comment:
Mainly because this pattern  would break the automatic doc generation, the loading property will no longer be considered ass public property from the POV of the doc generator algorithm.